.github/workflows/deploy.yml

```diff
@@ -1,18 +1,121 @@
 name: 'Deploy Application'
 
+
 on:
   workflow_call:
     inputs:
+      default-agent:
+        required: true
```

Comment:
You can leave it as `false`, this way the engineer won't need to configure it unless it's highly necessary